arch/src/x86_64/regs.rs

```diff
@@ -239,6 +246,11 @@ fn setup_page_tables(mem: &GuestMemoryMmap, sregs: &mut kvm_sregs) -> Result<()>
     sregs.cr3 = PML4_START.raw_value();
     sregs.cr4 |= X86_CR4_PAE;
     sregs.cr0 |= X86_CR0_PG;
+
+    if CpuidPatch::get_cpuid_reg(cpuid, 0x7, None, CpuidReg::ECX) == 0x10000 {
+        sregs.cr4 |= X86_CR4_LA57;
```

Comment:
Does that work with direct kernel boot with PVH kernels? AFAICT, the PVH kernel boot path does not set the LA57 bit.

Reply:
It doesn't work when we try to set the CR4.LA57 bit with PVH. We need to find the solution for this.